RV-Monitor/Kernel_Safety_Claims_by_Runtime_Verification_Monitors.md

```diff
@@ -0,0 +1,542 @@
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >beating
+   </td>
+   <td rowspan="2" >The ioctl returned success and the i6300 watchdog is ticking according to the programmed timeout
+   </td>
+   <td>write: the Safety App invoked a write syscall on the i6300 file descriptor
+   </td>
+   <td>reload
+   </td>
+  </tr>
+  <tr>
+   <td>BPA / unlock_wr: the watchdog timer registers were attempted to be written or a thread different from the safety app tried to unlock the i6300 HW for write access
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >reload
+   </td>
+   <td rowspan="3" >The watchdog is reloaded according to the current timeout and starts ticking
+   </td>
+   <td>failure: something went wrong when reloading the watchdog
+   </td>
+   <td>beating
+   </td>
+  </tr>
+  <tr>
+   <td>success: the reload operation succeeds and write is going to return success
+   </td>
+   <td>reload_check
+   </td>
+  </tr>
+  <tr>
+   <td>BPA / unlock_wr: the watchdog timer registers were attempted to be written or a thread different from the safety app tried to unlock the i6300 HW for write access
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >reload_check
+   </td>
+   <td rowspan="3" >Before write returns success, the RV monitor checks if the operations on the HW have been performed as expected 
+   </td>
+   <td>success: the operations on the HW have been performed as expected
+   </td>
+   <td>beating
+   </td>
+  </tr>
+  <tr>
+   <td>failure: write was going to return success while one or more operations on the HW were not as expected   
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td>BPA / unlock_wr: the watchdog timer registers were attempted to be written or a thread different from the safety app tried to unlock the i6300 HW for write access
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td>fail
+   </td>
+   <td>The monitored code is in an invalid state, hence the RV Monitor triggers Kernel Panic here 
+   </td>
+   <td>None
+   </td>
+   <td>
+   </td>
+  </tr>
+</table>
+
+
+
+# Limitations and Improvements
+
+
+## Tracepoints
+
+**TODO**: evaluate if it is safer OR easier (in terms of FuSa effort) to resolve monitoring points at compile time instead of using runtime tracepoints.
+
+The concern is that relying on runtime tracepoints imply allocating safety requirements to a mechanisms that is more complex than relying on compile time resolution of monitoring points hooks
+
+
+## Monitored Data access
+
+**TODO**: evaluate if, for failure avoidance purposes and hence availability purposes, it is doable to directly read the data to be monitored from the monitored code heap or stack instead of passing it as parameters on the stack. The rationale is that passing critical data through function parameters would increase the surface of possible corruption.
```

Comment:
... corruption. Which is not relevant from integrity perspective, but it is from availability perspective.

Reply:
right, it is written on top "and hence availability purposes" ....

Reply:
ok